### PR TITLE
Hw26.2

### PR DESCRIPTION
**Pull request by tarkaiev**

### CODE REVIEW

File: src/main/java/ecommerce/controllers/CompleteOrderController.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    private final OrderService orderService
+            = (OrderService) injector.getInstance(OrderService.class);
+    private final ShoppingCartService shoppingCartService
+            = (ShoppingCartService) injector.getInstance(ShoppingCartService.class);
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        Long shoppingCartId = Long.parseLong(req.getParameter("shoppingCartId"));
+        Order order = orderService.completeOrder(shoppingCartService.get(shoppingCartId));
```
Comment (julia-shvabovich) [inline src/main/java/ecommerce/controllers/CompleteOrderController.java]: I was told to replace this get() on getByUserId(). As I get it, in some cases your result won't be correct

File: src/main/java/ecommerce/controllers/CurrentCartController.java [hunk 2]
```diff
@@ -25,6 +25,7 @@ protected void doGet(HttpServletRequest req, HttpServletResponse resp)
         ShoppingCart currentShoppingCart = shoppingCartService.getByUserId(USER_ID);
         List<Product> products = currentShoppingCart.getProducts();
         req.setAttribute("products", products);
+        req.setAttribute("shoppingCartId", shoppingCartService.getByUserId(1L).getId());
```
Comment (julia-shvabovich) [inline src/main/java/ecommerce/controllers/CurrentCartController.java]: You have a constant field for user id

Comment (tarkaiev) [reply src/main/java/ecommerce/controllers/CurrentCartController.java]: lol, I do
will fix right now

File: src/main/java/ecommerce/controllers/CompleteOrderController.java (hunk 1 above)
Comment (tarkaiev) [reply src/main/java/ecommerce/controllers/CompleteOrderController.java]: done

File: src/main/java/ecommerce/controllers/CompleteOrderController.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    private final OrderService orderService
+            = (OrderService) injector.getInstance(OrderService.class);
+    private final ShoppingCartService shoppingCartService
+            = (ShoppingCartService) injector.getInstance(ShoppingCartService.class);
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws IOException {
+        Long shoppingCartId = Long.parseLong(req.getParameter("shoppingCartId"));
+        Order order = orderService.completeOrder(shoppingCartService.getByUserId(shoppingCartId));
```
Comment (dmytro-hryhoruk) [inline src/main/java/ecommerce/controllers/CompleteOrderController.java]: method with the name getByUserId() takes shopping cart id?

File: src/main/java/ecommerce/controllers/CurrentCartController.java [hunk 4]
```diff
@@ -12,7 +12,7 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-@WebServlet("/current/cart")
+@WebServlet("/cart/current")
 public class CurrentCartController extends HttpServlet {
```
Comment (dmytro-hryhoruk) [inline src/main/java/ecommerce/controllers/CurrentCartController.java]:
```suggestion
public class GetUserCartController extends HttpServlet {
```

File: src/main/java/ecommerce/controllers/CurrentCartController.java [hunk 5]
```diff
@@ -25,6 +25,7 @@ protected void doGet(HttpServletRequest req, HttpServletResponse resp)
         ShoppingCart currentShoppingCart = shoppingCartService.getByUserId(USER_ID);
         List<Product> products = currentShoppingCart.getProducts();
         req.setAttribute("products", products);
+        req.setAttribute("shoppingCartId", shoppingCartService.getByUserId(USER_ID).getId());
```
Comment (dmytro-hryhoruk) [inline src/main/java/ecommerce/controllers/CurrentCartController.java]: you are calling database twice. Let's save shoppingCart to local variable

File: src/main/java/ecommerce/controllers/GetOrderDetailsController.java [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        Long orderId = Long.parseLong(req.getParameter("id"));
+        Order order = orderService.get(orderId);
+        req.setAttribute("order", order);
+        double totalSum = 0.0;
+        for (Product p : order.getProducts()) {
+            totalSum += p.getPrice();
+        }
```
Comment (dmytro-hryhoruk) [inline src/main/java/ecommerce/controllers/GetOrderDetailsController.java]: let's make a method in orderService for this

**Review by dmytro-hryhoruk — COMMENTED** [review]
I can't see a second controller for orders (GetUserOrdersController)
https://mate-academy.github.io/jv-program-fulltime/02_jdbc_and_servlets/homework/lesson_16_hw2.html

File: src/main/java/ecommerce/controllers/CurrentCartController.java (hunk 4 above)
Comment (tarkaiev) [reply src/main/java/ecommerce/controllers/CurrentCartController.java]: done

File: src/main/java/ecommerce/controllers/CurrentCartController.java (hunk 5 above)
Comment (tarkaiev) [reply src/main/java/ecommerce/controllers/CurrentCartController.java]: done

File: src/main/java/ecommerce/controllers/CompleteOrderController.java (hunk 3 above)
Comment (tarkaiev) [reply src/main/java/ecommerce/controllers/CompleteOrderController.java]: fixed

File: src/main/java/ecommerce/controllers/GetOrderDetailsController.java (hunk 6 above)
Comment (tarkaiev) [reply src/main/java/ecommerce/controllers/GetOrderDetailsController.java]: done

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request merged**